### PR TITLE
Elm local install support

### PR DESCRIPTION
**Pull request by soywod**
Added here some support for local installation of Elm (for eg: via npm). All this code has been copied (and ajusted) from the Flow linter, so I don't think it is necessary to add some tests here.

### CODE REVIEW

File: ale_linters/elm/make.vim [hunk 1]
```diff
@@ -1,6 +1,15 @@
-" Author: buffalocoder - https://github.com/buffalocoder
+" Author: buffalocoder - https://github.com/buffalocoder, soywod - https://github.com/soywod
 " Description: Elm linting in Ale. Closely follows the Syntastic checker in https://github.com/ElmCast/elm-vim.
 
+call ale#Set('elm_make_executable', 'elm-make')
+call ale#Set('elm_make_use_global', 1)
```
Comment (w0rp) [inline ale_linters/elm/make.vim]: Change the default to `0`, to match the other linters. We should always prefer using a version more appropriate for a project by default.

File: ale_linters/elm/make.vim [hunk 2]
```diff
@@ -63,14 +73,16 @@ function! ale_linters#elm#make#GetCommand(buffer) abort
     " a sort of flag to tell the compiler not to generate an output file,
     " which is why this is hard coded here.
     " Source: https://github.com/elm-lang/elm-make/blob/master/src/Flags.hs
-    let l:elm_cmd = 'elm-make --report=json --output='.ale#Escape('/dev/null')
+    let l:elm_cmd = ale#Escape(l:elm_exe)
+    \   . ' --report=json'
+    \   . ' --output=' . ale#Escape('/dev/null')
```
Comment (w0rp) [inline ale_linters/elm/make.vim]: Use `g:ale#util#nul_file` here instead, as `/dev/null` doesn't work on Windows. (Except for that one Go program...)

**Comment by w0rp** [comment]
Make sure to document the new options too.

File: ale_linters/elm/make.vim (hunk 1 above)
Comment (soywod) [reply ale_linters/elm/make.vim]: Done

File: ale_linters/elm/make.vim (hunk 2 above)
Comment (soywod) [reply ale_linters/elm/make.vim]: Done

File: ale_linters/elm/make.vim (hunk 1 above)
Comment (w0rp) [reply ale_linters/elm/make.vim]: I don't see the changes yet. You might need to push a commit.

Comment (soywod) [reply ale_linters/elm/make.vim]: It's just for me, to not forget any task. It's done in local, I'll push soon

**Comment by w0rp** [comment]
There's a conflict to resolve here, from merging the other PR. I might fix this later.

**Comment by soywod** [comment]
I resolve my conflict here ;)

**Comment by w0rp** [comment]
Ah, could you move the `elm-make` after the `elm-format` section and make sure they are in the same order in the table of contents? They are supposed to be mostly sorted by name. I might add some Bash code for catching that at some point.

**Comment by w0rp** [comment]
Don't forget the table of contents.

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**